app/mailers/activity_mailer.rb

```diff
@@ -5,16 +5,20 @@
 class ActivityMailer < ApplicationMailer
   helper ApplicationFormattersHelper
 
+  layout :choose_layout
+
   # Emails a recipient, informing him of an activity.
   #
   # @param [User] recipient The recipient of the email.
   # @param [Course::Notification|UserNotification] notification The notification to be made
   #   available to the view, accessible using +@notification+.
   # @param [String] view_path The path to the view which should be rendered.
-  def email(recipient, notification, view_path)
+  # @param [String] layout_path The filename in app/views/layouts which should be rendered.
+  def email(recipient, notification, view_path, layout_path = nil)
```

Comment:
Do you think we should go for keyword parameters instead? Since there are quite a few params now.

Reply:
Sounds good, I've switched it to keyword arguments and modified the function calls in the base notifier.